core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/TarUtils.java

```diff
@@ -161,26 +161,27 @@ public static void untar(File tar, File targetDir) throws IOException {
   }
 
   public static void untar(InputStream in, File targetDir) throws IOException {
-    final TarArchiveInputStream tarIn = new TarArchiveInputStream(in);
-    byte[] b = new byte[BUF_SIZE];
-    TarArchiveEntry tarEntry;
-    while ((tarEntry = tarIn.getNextTarEntry()) != null) {
-      final File file = new File(targetDir, tarEntry.getName());
-      if (tarEntry.isDirectory()) {
-        if (!file.mkdirs()) {
-          throw new IOException("Unable to create folder " + file.getAbsolutePath());
-        }
-      } else {
-        final File parent = file.getParentFile();
-        if (!parent.exists()) {
-          if (!parent.mkdirs()) {
-            throw new IOException("Unable to create folder " + parent.getAbsolutePath());
+    try (TarArchiveInputStream tarIn = new TarArchiveInputStream(in)) {
```

Comment:
I think the idea might be in a fact that caller of untar method suppose to close ```InputStream``` for example https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/TarUtils.java#L158-L160